### PR TITLE
Adding SSH Keypair Instructions

### PR DESCRIPTION
**Pull request by SoftwareDev**

### CODE REVIEW

File: SSH_Keypair_Instructions.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Linux / MacOS instructions
+## Step 1: Generate a new SSH key for MarkUs
+
+1. Open up your Terminal and copy and paste the following text below. If you wish, you can add a label for your key pair. (ex: Laptop, School Computer, etc)  
+`ssh-keygen -t rsa -C "LABEL GOES HERE"`
```
Comment (david-yz-liu) [inline SSH_Keypair_Instructions.md]: Use `>` to distinguish commands from output

File: SSH_Keypair_Instructions.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+## Step 2: Add your key to your ssh-agent
+An **ssh-agent** is a tool which keeps track of your private & public key pairs and will help authenticate you when you try to establish connections.
+___
+1. Start up your ssh-agent by typing this into your terminal:  
+`eval $(ssh-agent -s)`  
+Or, if that does not work try:  
+`ssh-agent -s`  
+You should see something like if the command is successful:  
+`Agent pid 99887`
```
Comment (david-yz-liu) [inline SSH_Keypair_Instructions.md]: Clarify the pid doesn't matter

File: SSH_Keypair_Instructions.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+`Agent pid 99887`  
+2. Now add your newly generated key to the ssh-agent by running this command:  
+`ssh-add ~/.ssh/id_rsa`  
+You should see something like:  
+`Identity added: .../<yourUsername>/.ssh/id_rsa`  
+`(.../<yourUsername>/.ssh/id_rsa)`  
+
+## Step 3: Add your public key to MarkUs
+1. Adding your public key to MarkUs is done by visiting this page and clicking “**New Key Pair**”  
+2. Now you can choose to upload the public key file itself (located in the hidden folder in your home directory _**~/.ssh/id_rsa.pub**_ or by executing the following commands to copy and paste your public key contents:
```
Comment (david-yz-liu) [inline SSH_Keypair_Instructions.md]: I'd probably use `path/to/file` style rather than bold and italics

File: SSH_Keypair_Instructions.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+`Identity added: .../<yourUsername>/.ssh/id_rsa`  
+`(.../<yourUsername>/.ssh/id_rsa)`  
+
+## Step 3: Add your public key to MarkUs
+1. Adding your public key to MarkUs is done by visiting this page and clicking “**New Key Pair**”  
+2. Now you can choose to upload the public key file itself (located in the hidden folder in your home directory _**~/.ssh/id_rsa.pub**_ or by executing the following commands to copy and paste your public key contents:  
+`cat ~/.ssh/id_rsa.pub`  
+Then you can copy and paste your entire key into MarkUs:  
+![Copying the output to the clipboard from running the "cat" program on your public key](http://alexgrenier.ca/MarkUs/1.png)  
+![Copy paste the results of running the "cat" program on your public key and paste it in the MarkUs "Add a Key-pair" form.](http://alexgrenier.ca/MarkUs/2.png)
```
Comment (david-yz-liu) [inline SSH_Keypair_Instructions.md]: Does this picture appear for you? Seemed like a hyperlink for me, for some reason

File: SSH_Keypair_Instructions.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+
+***
+
+# Windows instructions
+## Step 1: Install Git
+
+1. Download and install Git on your machine. http://git-scm.com/downloads
```
Comment (david-yz-liu) [inline SSH_Keypair_Instructions.md]: Make Git a hyperlink

File: SSH_Keypair_Instructions.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+# Windows instructions
+## Step 1: Install Git
+
+1. Download and install Git on your machine. http://git-scm.com/downloads  
+2. Open **Git bash** by clicking the start menu and typing in “git bash”.  
+  
+**Windows 7**  
+ 
+![Search for "git bash" when you open your Start Menu in Windows 7](http://alexgrenier.ca/MarkUs/3.png)
```
Comment (david-yz-liu) [inline SSH_Keypair_Instructions.md]: Same with this picture and the next.

File: SSH_Keypair_Instructions.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+`ssh-add ~/.ssh/id_rsa`  
+You should see something like:  
+`Identity added: .../<yourUsername>/.ssh/id_rsa`  
+`(.../<yourUsername>/.ssh/id_rsa)`  
+_**From now on, continue to use the Git bash for your Git commands on your Windows machine.**_
+
+## Step 4: Add your public key to MarkUs  
+
+1. Adding your public key to MarkUs is done by visiting this page and clicking “**New Key Pair**”  
+2. Now you can choose to upload the public key file itself (located in the hidden folder in your home directory _**~/.ssh/id_rsa.pub**_ or by executing the following commands from within your **Git bash** to copy and paste your public key contents:
```
Comment (david-yz-liu) [inline SSH_Keypair_Instructions.md]: This should be clarified for Windows.

**Comment by david-yz-liu** [comment]
Nice write-up. I think the only major complaint I have is that it seems like there's a lot of duplicated instructions between the two OSes, and I wonder is perhaps you should just merge them, and point out the OS-specific parts when the arise. E.g., the extra step of installing Git on Windows, locations of the "Home" directories.

**Comment by SoftwareDev** [comment]
@david-yz-liu Solid points, I'll get on that. I agree about the duplication part... I just wanted to really make things clear for students who are not comfortable with key generation. But I'll see what I can do.

**Comment by david-yz-liu** [comment]
@SoftwareDev Perhaps make two separate pages then - one for Windows and one for OSX/Linux. We can link to both from MarkUs.

**Comment by SoftwareDev** [comment]
@david-yz-liu Good idea, I'll do that

**Pull request merged**